### DISCUSSION TITLE
Creating volumes and buckets using python's requests library.

### DISCUSSION THREAD

**Discussion by kwstarikas** [apache/ozone#6204] [General]
Hello community. I am running ozone locally, and i use boto3 to make requests(create/update/delete/get buckets and  keys). How can i create volumes. I try using the requests library by sending a post request to om_url/volume but i get 404 error . (its my first discussion so sorry if this post is confusing.)

**Comment by adoroszlai** [comment]
Hi @kwstarikas,

Volumes can be created via Ozone command-line client: `ozone sh volume create`.

S3 client cannot be used to create volumes, because it is not aware of this part of the namespace (it does not exist in S3).

**Reply by kwstarikas** [reply]
Yes i can create volumes via the OM's terminal with this command. But how do i create a volume with the python's requests library? i want to find a way to make an endpoint that creates volumes in my Ozone.

**Reply by devmadhuu** [reply]
As far as I know, ozone don't expose direct http end point for creating volumes, buckets, but if you want you can create your own API endpoint using ozone client library for your cluster.

**Reply by kerneltime** [reply]
@kwstarikas this looks like a good feature request, you can create an upstream Jira https://issues.apache.org/jira/issues/ to post your thoughts and requirements.
